components/molecule/photoUploader/src/photoTools.js

```diff
@@ -0,0 +1,317 @@
+            return cropAndRotateImage({
+              base64Image,
+              ...options
+            })
+          })
+          .then(value => base64ToBlob(value))
+          .then(({blob, base64}) => {
+            resolve({
+              file,
+              blob,
+              originalBase64,
+              croppedBase64: base64
+            })
+          })
+          .catch(e => {
+            resolve({hasErrors: true})
+          })
+      }
+    })
+  } else {
+    return new Promise((resolve, reject) => {
+      cropAndRotateImage({
+        imageURL: url,
+        ...options
+      })
+        .then(value => base64ToBlob(value))
+        .then(({blob, base64}) => {
+          resolve({
+            url,
+            blob,
+            croppedBase64: base64
+          })
+        })
+        .catch(e => {
+          resolve({url, hasErrors: true})
+        })
+    })
+  }
+}
+
+export function resizeImage({
+  base64Image,
+  exifOrientation,
+  maxImageWidth,
+  maxImageHeight,
+  imageMimeType
+}) {
+  const canvas = document.createElement('canvas')
+  const context = canvas.getContext('2d')
+
+  const image = new Image() // eslint-disable-line
+  image.src = base64Image
+
+  console.log('exif oritentation : ', exifOrientation)
+
+  return new Promise((resolve, reject) => {
+    image.onerror = () => {
+      reject(new Error(`Image load fails. Origin: File resizing`))
+    }
+    image.onload = () => {
+      const inputWidth = image.naturalWidth
+      const inputHeight = image.naturalHeight
+
+      if (inputHeight > maxImageHeight || inputWidth > maxImageWidth) {
+        const inputImageAspectRatio = inputWidth / inputHeight
+
+        if (inputHeight > inputWidth) {
+          const scaleFactorHeight = maxImageHeight / inputHeight
+
+          canvas.width = maxImageHeight * inputImageAspectRatio
+          canvas.height = maxImageHeight
+
+          context.translate(canvas.width / 2, canvas.height / 2)
+          context.scale(scaleFactorHeight, scaleFactorHeight)
+          context.translate(-canvas.width / 2, -canvas.height / 2)
+        } else {
+          const scaleFactorWidth = maxImageWidth / inputWidth
+
+          canvas.width = maxImageWidth
+          canvas.height = maxImageWidth / inputImageAspectRatio
+
+          context.translate(canvas.width / 2, canvas.height / 2)
+          context.scale(scaleFactorWidth, scaleFactorWidth)
+          context.translate(-canvas.width / 2, -canvas.height / 2)
+        }
+        context.drawImage(
+          image,
+          (canvas.width - inputWidth) / 2,
+          (canvas.height - inputHeight) / 2
+        )
+        resolve(canvas.toDataURL(imageMimeType, 1))
+      } else {
+        resolve(base64Image)
+      }
+    }
+  })
+}
+
+export function cropAndRotateImage({
+  base64Image,
+  imageURL,
+  rotation = 0,
+  outputImageAspectRatio,
+  maxImageHeight,
+  maxImageWidth,
+  imageQuality = 0.8,
+  imageMimeType = 'image/jpeg'
```

Comment:
have seen this assignment at least a couple of times here so it might make sense to have a const somewhere for `image/jpeg`, don't you think?

Reply:
Done! I also set a const to imageQuality default value! Thanks!